playbooks/allocate_pubcloud.yml

```diff
@@ -9,6 +9,7 @@
     inventory_path: "{{ lookup('env', 'WORKSPACE') }}/rpc-gating/playbooks/inventory"
     user_data_path: "{{ lookup('env', 'WORKSPACE') }}/rpc-gating/scripts/user_data_pubcloud.sh"
     cloud_name: "public_cloud"
+    boot_timeout: "{{ lookup('env', 'BOOT_TIMEOUT') | default(900, true) }}"
```

Comment:
What does `true` do? I couldn't find a reference to a second argument in the Ansible documentation for `default`.

Reply:
From http://jinja.pocoo.org/docs/dev/templates/#default:

```This will output the value of my_variable if the variable was defined, otherwise 'my_variable is not defined'. If you want to use default with variables that evaluate to false you have to set the second parameter to true```

If I don't add the true, `boot_timeout` defaults to `""`:

```root@aio1:/opt/openstack-ansible/playbooks# cat test.yml
- hosts: localhost
  connection: local
  gather_facts: False
  vars:
    boot_timeout: "{{ lookup('env', 'BOOT_TIMEOUT') | default(900) }}"
  tasks:
    - name: Create primary region list
      debug: var=boot_timeout
root@aio1:/opt/openstack-ansible/playbooks# openstack-ansible test.yml
Variable files: "-e @/etc/openstack_deploy/user_secrets.yml -e @/etc/openstack_deploy/user_variables.yml "
 [WARNING]: Unable to parse /etc/openstack_deploy/inventory.ini as an inventory source


PLAY [localhost] *************************************************************************************************************************************************************************************************************************************************************

TASK [Create primary region list] ********************************************************************************************************************************************************************************************************************************************
Tuesday 06 March 2018  14:58:28 +0000 (0:00:00.226)       0:00:00.226 *********
ok: [localhost] => {
    "boot_timeout": ""
}

PLAY RECAP *******************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0

Tuesday 06 March 2018  14:58:28 +0000 (0:00:00.043)       0:00:00.270 *********
===============================================================================
Create primary region list -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- 0.04s

EXIT NOTICE [Playbook execution success] **************************************
===============================================================================
root@aio1:/opt/openstack-ansible/playbooks#```

Reply:
thanks